example/lib/widgets/main_button.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+
+import '../utils/styles.dart';
+import '../utils/dimensions.dart';
+
+class MainButton extends StatelessWidget {
+  const MainButton({
+    Key? key,
+    required this.onPressedFunction,
+    required this.children,
+  }) : super(key: key);
+  final void Function()? onPressedFunction;
+  final List<Widget> children;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      style: ButtonStyle(
+        backgroundColor: MaterialStateProperty.resolveWith(
+          (states) {
+            if (states.contains(
+              MaterialState.disabled,
+            ))
+              return Colors.purple.shade900.withOpacity(
+                Styles.colorOpacity800,
+              );
+            return Colors.purple.shade500;
+          },
+        ),
+        foregroundColor: MaterialStateProperty.resolveWith(
+          (states) {
+            if (states.contains(
+              MaterialState.disabled,
+            ))
+              return Colors.grey.shade500.withOpacity(
+                Styles.colorOpacity900,
+              );
+            return Colors.grey.shade200;
+          },
+        ),
+      ),
+      onPressed: onPressedFunction != null
+          ? () {
+              onPressedFunction!();
+            }
+          : null,
```

Comment:
Instead of this, use the logic that you have on location_page
isInitialized ? onPressedFunction() : (){}, try to not assign null

Reply:
Done